diff --git a/linux/Config.in b/linux/Config.in
index 02c39b3..95623a0 100644
--- a/linux/Config.in
+++ b/linux/Config.in
@@ -370,8 +370,16 @@ config BR2_LINUX_KERNEL_INTREE_DTS_NAME
 	  the trailing .dts. You can provide a list of
 	  dts files to build, separated by spaces.
 
+config BR2_LINUX_KERNEL_USE_CUSTOM_DTS
+	bool "Use a custom device tree file"
+	help
+	  Use a custom device tree file, i.e, a device
+	  tree file that does not belong to the kernel
+	  source tree.
+
 config BR2_LINUX_KERNEL_CUSTOM_DTS_PATH
 	string "Out-of-tree Device Tree Source file paths"
+	depends on BR2_LINUX_KERNEL_USE_CUSTOM_DTS
 	help
 	  Path to to out-of-tree device tree source files.
 	  You can provide a list of dts paths to copy and
